web/src/main/java/se/inera/statistics/web/service/task/TaskCoordinatorServiceImpl.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (C) 2023 Inera AB (http://www.inera.se)
+ *
+ * This file is part of sklintyg (https://github.com/sklintyg).
+ *
+ * sklintyg is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * sklintyg is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package se.inera.statistics.web.service.task;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.cache.Cache;
+import org.springframework.stereotype.Component;
+
+@Component()
+public class TaskCoordinatorServiceImpl implements TaskCoordinatorService {
+
+    private static final Logger LOG = LoggerFactory.getLogger(TaskCoordinatorServiceImpl.class);
+
+    private final int simultaneousCallsAllowed;
+    private final Cache taskCoordinatorCache;
+    private final ObjectMapper objectMapper;
+
+    public TaskCoordinatorServiceImpl(
+        @Value("${taskCoordinatorService.simultaneous.calls.allowed}") int simultaneousCallsAllowed,
+        @Qualifier("taskCoordinatorRedisCache") Cache taskCoordinatorCache,
+        ObjectMapper objectMapper) {
+        this.simultaneousCallsAllowed = simultaneousCallsAllowed;
+        this.taskCoordinatorCache = taskCoordinatorCache;
+        this.objectMapper = objectMapper;
+    }
+
+    @Override
+    public TaskCoordinatorResponse request(String userHsaId) {
+        final var cachedRequests = getCachedRequests(userHsaId);
+        if (limitForSimultaneousCallsExceeded(cachedRequests)) {
+            LOG.debug(
```

Comment:
Maybe change this to info instead of debug. Otherwise we probably won't see this valuable log in production.

Reply:
Fixed!